internal/metastore/model/segment_index.go

```diff
@@ -32,7 +38,10 @@ func UnmarshalSegmentIndexModel(segIndex *indexpb.SegmentIndex) *SegmentIndex {
 	if segIndex == nil {
 		return nil
 	}
-
+	memSize := segIndex.MemSize
+	if memSize == 0 {
```

Comment:
If the mem size in meta is zero, use serialized size with Multiplier to estimate the mem size.